src/jit/codegencommon.cpp

```diff
@@ -3407,16 +3398,19 @@ void CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg, bool* pXtraRegClobbere
             {
                 return varDsc.GetHfaType();
             }
-            return varDsc.lvType;
+            return compiler->mangleVarArgsType(varDsc.lvType);
```

Comment:
Since HFA types are also mangled by th varArg calling convention shouldthis be:

```
             var_type normalType = varDsc.lvType;
             // Check if this is an HFA register arg and return the HFA type
             if (varDsc.lvIsHfaRegArg())
             {
                 normalType = varDsc.GetHfaType();
             }
             return compiler->mangleVarArgsType(normalType);
```

Reply:
or add
`assert(!compiler->info.compIsVarArgs);`

Reply:
Also add a comment explaining what mangleVarArgsType migt do here, and on what targets it is active on.

Reply:
Looks like there are several places this hfa decision will need to be made. I think that it is worth its own change.

Reply:
I thought for varargs, we didn't get any HFA? That is, shouldn't the incoming args not be treated as HFA in the first place?

Reply:
This is a true statement, which is why the assert is the better path.